src/lib.rs

```diff
@@ -557,9 +586,16 @@ pub struct Connection {
 
 #[derive(Serialize, Clone, Debug)]
 enum AuthStyle {
-    //    Credentials(String, String),
+    /// Authenticate using a token.
     Token(String),
+
+    /// Authenticate using a username and password.
     UserPass(String, String),
+
+    /// Authenticate using a `.creds` file.
+    Credentials(PathBuf),
```

Comment:
This in the Go client is simply and concatenation of JWT and pl(nkey) which we will want to also support as separate ones. Look at Go client for pointers.

Reply:
Done - this variant now looks like this:
```rust
/// Authenticate using a `.creds` file.
Credentials {
    /// Securely loads the user JWT.
    jwt_cb: Arc<dyn Fn() -> io::Result<String> + Send + Sync>,
    /// Securely loads the nkey and signs the nonce passed as an argument.
    sig_cb: Arc<dyn Fn(&[u8]) -> io::Result<SecureString> + Send + Sync>,
},
```

Reply:
Looks good!